src/parser/solar.rs

```diff
@@ -411,6 +425,82 @@ trait Extract {
     fn extract_definition(self, item: &Item, visitor: &mut LintspecVisitor) -> Option<Definition>;
 }
 
+impl Extract for &solar_parse::ast::ItemContract<'_> {
+    fn extract_definition(self, item: &Item, visitor: &mut LintspecVisitor) -> Option<Definition> {
+        let name = self.name.to_string();
+
+        let (natspec, span) = match extract_natspec(&item.docs, visitor, &[]) {
+            Ok(extracted) => extracted.map_or_else(
+                || {
+                    let span = match &item.kind {
+                        ItemKind::Contract(item_contract) => {
+                            let end_bases = item_contract
+                                .bases
+                                .last()
+                                .map_or(item_contract.name.span.hi(), |b| b.span().hi());
+                            let end_specifiers = item_contract
+                                .layout
+                                .as_ref()
+                                .map_or(item_contract.name.span.hi(), |l| l.span.hi());
+                            visitor
+                                .span_to_textrange(item.span.with_hi(end_bases.max(end_specifiers)))
+                        }
+                        _ => visitor.span_to_textrange(item.span),
```

Comment:
is this reachable?

Reply:
e: checked the main solar.rs, I think it's dead code I introduced in the other extract_definition (my bad) - down to PR a fix or you push it here, as you prefer!